File: app/src/main/java/com/education4all/NotificationReceiver.java
package com.education4all;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;

import androidx.annotation.RequiresApi;

import com.education4all.utils.Utils;

public class NotificationReceiver extends BroadcastReceiver {


    @RequiresApi(api = Build.VERSION_CODES.M)
    @Override
    public void onReceive(Context context, Intent intent) {
        NotificationHelper notificationHelper = new NotificationHelper(context);

        String action = intent.getAction();
        switch (action) {
            case NotificationHelper.CLOSE:
                notificationHelper.cancel(NotificationHelper.NOTIFICATION_ID);
                break;
            case NotificationHelper.DELAY:
                notificationHelper.delay();
                notificationHelper.cancel(NotificationHelper.NOTIFICATION_ID);
                break;
            case NotificationHelper.MAKE:
                if (!Utils.reachedGoal(context))
                    notificationHelper.createNotification();
                break;
        }
        notificationHelper.repeat();
    }
}

File: app/src/main/java/com/education4all/NotificationReceiver.kt
package com.education4all

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.education4all.utils.Utils

class NotificationReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val notificationHelper = NotificationHelper(context)
        val action = intent.action
        when (action) {
            NotificationHelper.Companion.CLOSE -> {
                notificationHelper.cancel(NotificationHelper.Companion.NOTIFICATION_ID)
                notificationHelper.repeat()
            }
            NotificationHelper.Companion.DELAY -> {
                notificationHelper.cancel(NotificationHelper.Companion.NOTIFICATION_ID)
                notificationHelper.delay()
            }
            NotificationHelper.Companion.MAKE -> {
                if (!Utils.reachedGoal(context)) notificationHelper.createNotification()
                notificationHelper.repeat()
            }
            NotificationHelper.Companion.MAKEDELAYED -> if (!Utils.reachedGoal(context)) notificationHelper.createNotification()
        }
    }
}